fabric/src/main/java/net/minescript/common/Minescript.java

```diff
@@ -3981,6 +4010,116 @@ private static Optional<String> handleScriptFunction(
         }
         return Optional.of(toJsonString(getScreenName().orElse(null)));
 
+      case "container_get_item": // Item in slot in container
+      {
+        if (args.size() != 1) {
+          logUserError("Error: `{}` expected 1 param (slot id) but got: {}", functionName, argsString);
+          return Optional.empty();
+        }
+        Screen screen = minecraft.currentScreen;
+        int slot = getStrictIntValue(args.get(0)).getAsInt();
+        if (screen == null) {
+          logUserError("Error: `{}`: slot {} does not exist", functionName, slot);
+          return Optional.empty();
+        }
+        if (screen instanceof HandledScreen<?>) {
+          HandledScreen handledScreen = (HandledScreen) screen;
+          ScreenHandler screenHandler = handledScreen.getScreenHandler();
+          if (screenHandler.slots.size() <= slot) {
+            return Optional.empty();
+          }
+          ItemStack itemStack = screenHandler.slots.get(slot).getStack();
+          return Optional.of(itemStackToJsonString(itemStack, OptionalInt.of(slot), false));
+        } else {
+          return Optional.empty();
+        }
+      }
+
+      case "container_get_items": // List of Items in Chest
+      {
```

Comment:
Please `throw new IllegalArgumentException("Expected no params but got: " + argsString)` if `!args.isEmpty()`. See `"register_key_event_listener"` for an example.

Translation of Java exceptions to Python exceptions is new in v3.2 and should be preferred for new script functions that encounter exceptional conditions.